[gdpa] subtiling activation to reduce register pressure

manman-ren opened the pull request:
Summary: use vector fma etc use new way of calculating fast_gelu * scale

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

davidberard98 committed:
Support old triton versions w/ setup.py in ./python (#259)

Summary:
I want to ab test triton 3.3 and triton 3.4 to look for perf regressions or failures not caught in PyTorch CI. But Triton 3.3 is too old for the existing setup script, which assumes setup.py in the ./ directory (in triton 3.3, it's in ./python/setup.py).

This PR adds support to build either version.

before (fails to compile 3.3): https://github.com/pytorch-labs/tritonbench/actions/runs/16012761341
after (succeeds in compiling 3.3): https://github.com/pytorch-labs/tritonbench/actions/runs/16013163616

Pull Request resolved: https://github.com/pytorch-labs/tritonbench/pull/259

Reviewed By: xuzhao9

Differential Revision: D77639239

Pulled By: davidberard98

fbshipit-source-id: 8c21e15284ccb73b04d4242e87f192fce87a6389

xuzhao9 committed:
Fix broken matmul test case

Summary:
To fix https://www.internalfb.com/intern/test/844425154028649?ref_report_id=0

It turns out that we can skip an entire operator by just adding it to the skip yaml file.

Differential Revision: D77670538

fbshipit-source-id: 3f2fd54d84974f83fe49fb4dc9edee828a78e9e5

xuzhao9 committed:
Add more gemm configs (#261)

Summary:
Add more gemm run configs

Pull Request resolved: https://github.com/pytorch-labs/tritonbench/pull/261

Reviewed By: FindHao

Differential Revision: D77681038

Pulled By: xuzhao9

fbshipit-source-id: dee10b2645f7a40eaf3460cd6941d9818f6cc310

davidberard98 committed:
fix ab test: "side_b_commit" (#262)

Summary:
Previously both the "A Side" and "B Side" were running off the same Triton commit - this updates the "B Side" to actually use the side_b_commit instead of the side_a_commit

before this PR: https://github.com/pytorch-labs/tritonbench/actions/runs/16013163616 - go to "Compile Triton (Side B)" step and see that it's compiling release/3.3.x, while side_b_commit = release/3.4.x
after this PR: https://github.com/pytorch-labs/tritonbench/actions/runs/16035767306 - see "Compile Triton (Side B)" actually shows release/3.4.x

Pull Request resolved: https://github.com/pytorch-labs/tritonbench/pull/262

Reviewed By: xuzhao9

Differential Revision: D77687644

Pulled By: davidberard98

fbshipit-source-id: 29b9d4fda3bed5ad256588164cc1bd29f420903a

manman-ren committed:
Add code to enable autoWS from triton-lang (#260)

Summary:
Also clean up setting up configs for different optimization variants.
- Extract computation in inner loop as a helper function attn_fwd_iteration
- We have 3 different versions of attn_fwd_inner: attn_fwd_inner[_autows|_ws], _autows is for OSS triton where we use warp_specialize on ForOp to enable warpspec, _ws is for internal triton, where we use tl.async_task and num_consumer_groups to enable warpspec
- We have 2 different versions of attn_fwd_compute: attn_fwd_compute[_ws], attn_fwd_compute_ws calls attn_fwd_inner_ws, attn_fwd_compute calls either attn_fwd_inner or attn_fwd_inner_autows depending on whether warpspec should be enabled via argument WARP_SPECIALIZE
- We have 4 top-level kernels: _attn_fwd_ws, _attn_fwd_base_opt, _attn_fwd_tma_unified, _attn_fwd_tma_ws_persistent
- Depending on env var ENABLE_EXPLICIT_WS, _attn_fwd_ws will call either _attn_fwd_compute_ws or attn_fwd_compute with WARP_SPECIALIZE=True

```
TRITON_PRINT_AUTOTUNING=1 WITH_TMA=1 CUDA_VISIBLE_DEVICES=5 TORCH_CUDA_ARCH_LIST=9.0a ./denoise.sh python run.py --op flash_attention --only triton_tutorial_flash_v2_tma_ws_persistent,triton_tutorial_flash_v2_tma_ws --num-inputs 1 --seq-len 8192 --metrics tflops --batch 8 --n-heads 16 --d-head 128
  (Batch, Heads, SeqLen, SeqLen_KV, Dhead)    triton_tutorial_flash_v2_tma_ws_persistent-tflops    triton_tutorial_flash_v2_tma_ws-tflops
------------------------------------------  ---------------------------------------------------  ----------------------------------------
                  (8, 16, 8192, 8192, 128)                                              517.261                                   499.359
```

Pull Request resolved: https://github.com/pytorch-labs/tritonbench/pull/260

Reviewed By: xuzhao9

Differential Revision: D77639159

Pulled By: manman-ren

fbshipit-source-id: 9e9fdf3216fb74ecd215b3a4e222f21cba330ee4

danzimm committed:
Change version to 3.3.1+fb to pass packaging.version parsing

Summary: `vllm` parses the triton version e.g. [here](https://fburl.com/code/g9fofccn), but our new version label doesn't parse (it throws `InvalidVersion`). Let's make it parse. Shout out to xli who reported this and came up with the fix.

Reviewed By: xli

Differential Revision: D77699605

fbshipit-source-id: a05456acb9951714418528f9d77a6c0341135f6d

xuzhao9 committed:
Enable more tests (#251)

Summary:
BE efforts enables more tests in nightly.

Note that we can't enable a couple of fp8_* operators from FBGEMM, because FBGEMM Triton code still uses the old TMA API and therefore can't work on Triton main.

Pull Request resolved: https://github.com/pytorch-labs/tritonbench/pull/251

Test Plan: https://github.com/pytorch-labs/tritonbench/actions/runs/16037334650/job/45251972846?pr=251

Reviewed By: davidberard98

Differential Revision: D77697981

Pulled By: xuzhao9

fbshipit-source-id: 3704ea1d19ed973ae03b5d8e3693e0fb18a3b3b4

xuzhao9 committed:
Add single side test to benchmark (#255)

Summary:
Testing facebook's triton: https://github.com/facebookexperimental/triton

Pull Request resolved: https://github.com/pytorch-labs/tritonbench/pull/255

Test Plan: https://github.com/pytorch-labs/tritonbench/actions/runs/15936726775

Reviewed By: FindHao

Differential Revision: D77734402

Pulled By: xuzhao9

fbshipit-source-id: 40096410d1682d3e688f18209f5f9f04dbe6797b

FindHao committed:
Use liger-kernel-nightly (#265)

Summary:
Update liger kernel dependency to liger-kernel-nightly.

Pull Request resolved: https://github.com/pytorch-labs/tritonbench/pull/265

Reviewed By: xuzhao9

Differential Revision: D77746773

Pulled By: FindHao

fbshipit-source-id: c8d2b6005e58f042abad213ebe01b02dbbd0fa9e

xuzhao9 committed:
Allows operator to override accuacy checking function

Summary: We should allow each operator to customize its way to define accuacy.

Reviewed By: yf225

Differential Revision: D77750474

fbshipit-source-id: c0580642077beb0bf828fd28e8661dd425443698

yf225 committed:
Make vector_add Triton version return output tensor, to fix accuracy check

Summary: As titled. From Helion project side we'd like to reuse TritonBench as benchmarking infra, and making accuracy testing work is an important part of it.

Reviewed By: xuzhao9

Differential Revision: D77759603

fbshipit-source-id: 48943b0d18a348423b254090c92e992e5c0b933a

xuzhao9 committed:
Remove unused os import in low_mem_dropout operator

Summary:
## Instructions about RACER Diffs:
You will receive **FULL CREDIT** (ETS) for this diff if you either:
 - [**Recommended**]Commandeer and land this diff

Once landed, feel free to claim the associated task and the EYS project.

 This diff is removing the unused imports from Code Quality Score from [Monetization codehub](https://fburl.com/quality/58ozwacb):
 - Accept and ship the diff (racer do not do auto land)
 - [**Recommended**]Commandeer and land this diff.

If you are happy with the changes, please accept and ship, or commandeer it if minor edits are needed.
If you are not happy with the changes, please comment on the diff with clear actions and send it back to the author. Racer will pick it up and re-generage.
If you really feel the Racer is not helping with this change (alas, some complex changes are hard for AI) feel free to abandon this diff.

This diff was pre-created by Racer AI agent for your convenience on top of T229209355. More details about the program see this [post](https://fb.workplace.com/groups/1177219810659335/permalink/1198503531864296/). For questions or suggestions please post in [RACER Feedback and Q&A](https://fb.workplace.com/groups/2477474979269093) group.
## Summary:
Removed an unused 'os' import from the low_mem_dropout operator.py file as identified by the flake8 rule F401.
 ---
> Generated by [RACER](https://www.internalfb.com/wiki/RACER_(Risk-Aware_Code_Editing_and_Refactoring)/), powered by [Confucius](https://www.internalfb.com/wiki/Confucius/Analect/Shared_Analects/Confucius_Code_Assist_(CCA)/)
[Session](https://www.internalfb.com/confucius?session_id=a8813953-53e3-11f0-8aa9-9299bdf9c8b5&tab=Chat), [Trace](https://www.internalfb.com/confucius?session_id=a8813953-53e3-11f0-8aa9-9299bdf9c8b5&tab=Trace)

Reviewed By: FindHao

Differential Revision: D77481109

fbshipit-source-id: a2eefde307c14e67b9ac2242fae37357cb0f526f

manman-ren committed:
Add blackwell version of FA (#263)

Summary: Pull Request resolved: https://github.com/pytorch-labs/tritonbench/pull/263

Reviewed By: xuzhao9

Differential Revision: D77743107

Pulled By: manman-ren

fbshipit-source-id: 0e2d6209dfabbc3590d4b53dcc3fcbf81726ac9c

xuzhao9 committed:
Update submodules (#266)

Summary:
Updating the following submodules
- FBGEMM
- ThunderKittens
- aiter
- ~~flash_attention~~ flash_attention build failure, to investigate.
- ~~generative_recommenders~~  Can not update generative_recommenders because it uses 'triton.language.extra.cuda. experimental_device_tensormap_create2d' which does not exist in Triton main

Test build:
Check out CI

Pull Request resolved: https://github.com/pytorch-labs/tritonbench/pull/266

Reviewed By: FindHao

Differential Revision: D77884591

Pulled By: xuzhao9

fbshipit-source-id: 468f2324c8d03b92193332db3147ea7230a01fd8

xuzhao9 committed:
Update nightly test generation script (#267)

Summary:
We include an operator test in nightly if the following conditions `1 AND (2 OR 3)` are satisfied. This includes both fwd and bwd passes.

1. the operator has at least 1 Triton implementation
2. the operator has a baseline implementation to measure speedup
3. the operator has TFLOPS metric

Pull Request resolved: https://github.com/pytorch-labs/tritonbench/pull/267

Reviewed By: davidberard98

Differential Revision: D77873222

Pulled By: xuzhao9

fbshipit-source-id: f35a01fad77a2d5593717d70c349dfdfbed2dfc8

yf225 committed:
Add multi-blocks support to sum triton kernel, to enable large input size (#269)

Summary:
Currently, running `python run.py --op sum` fails with this error when tensor size is large:
```python
Traceback (most recent call last):
  File "/home/willfeng/local/pytorch-nightly/triton/language/core.py", line 34, in wrapper
    return fn(*args, **kwargs)
  File "/home/willfeng/local/pytorch-nightly/triton/language/core.py", line 1451, in arange
    return semantic.arange(start, end, _builder)
           ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^
  File "/home/willfeng/local/pytorch-nightly/triton/language/semantic.py", line 625, in arange
    ret_ty = tl.block_type(tl.int32, shape)
  File "/home/willfeng/local/pytorch-nightly/triton/language/core.py", line 648, in __init__
    self.numel = validate_block_shape(self.shape)
                 ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^
  File "/home/willfeng/local/pytorch-nightly/triton/language/_utils.py", line 20, in validate_block_shape
    raise ValueError(f"numel ({numel}) exceeds triton maximum tensor numel ({TRITON_MAX_TENSOR_NUMEL})")
ValueError: numel (2097152) exceeds triton maximum tensor numel (1048576)
```
This PR fixes the error by splitting M into multiple blocks when M's next power of 2 is too large.

Pull Request resolved: https://github.com/pytorch-labs/tritonbench/pull/269

Reviewed By: xuzhao9

Differential Revision: D77987069

Pulled By: yf225

fbshipit-source-id: 7505e6b3886102cd28a65ae3e6681379c963c544

xuzhao9 committed:
Add docs on metrics and installation (#271)

Summary:
We are gaining more OSS users, so we are adding more docs!

Pull Request resolved: https://github.com/pytorch-labs/tritonbench/pull/271

Reviewed By: FindHao

Differential Revision: D78026677

Pulled By: xuzhao9

fbshipit-source-id: e0d22e4c58d9b0acb30238ad259a481bf43fcf67

xuzhao9 committed:
Add tflops to default metrics

Summary: Make sure we print tflops metrics in the output.

Reviewed By: PaulZhang12

Differential Revision: D77450095

fbshipit-source-id: a7a70a0ab9dfd107603308885770d22cea40b5c6

xuzhao9 committed:
Add aten operator loader

Summary:
Refactor ATen operator loaders.

Move input configs from `operator_loaders` to `tritonbench/data/input_configs`.

Reviewed By: FindHao

Differential Revision: D77240210

fbshipit-source-id: ca40604a3ef5208d9d3edb0a9939ede200ea3a02

njriasan committed:
Include blackwell warpspec persistent matmul in TritonBench

Summary:
Adds the various blackwell matmuls that can support warpspec from https://github.com/triton-lang/triton/blob/main/python/tutorials/09-persistent-matmul.py to TritonBench so we benchmark them on OmniFm shapes.

Note: At this point I made the explicit choice to avoid making warp_spec an autotune configuration. The reasoning here is to both simplify benchmarking as we won't need to "determine" if/when warp spec is useful/how useful and because there might be bugs in warpspec.

Reviewed By: PaulZhang12

Differential Revision: D77053488

fbshipit-source-id: 9a3c3135e4646b71192b6e61d815d7043c7ee885

njriasan committed:
Fix metrics collection with --bypass-fail

Summary: Fixes an issue where computing the average with `--bypass-fail` will fail if a subset of the rows are None as it will revert `avg_row[col_num]` to None.

Reviewed By: xuzhao9

Differential Revision: D77949601

fbshipit-source-id: 3ff3d0491c4e3ff5ba2e652e4fad154067fa9707

xuzhao9 committed:
format code and fix bugs from input loader (#272)

Summary:
The Aten op diff broke OSS CI, fixing it.

Remove the old `operator_loader` files.

Pull Request resolved: https://github.com/pytorch-labs/tritonbench/pull/272

Reviewed By: FindHao

Differential Revision: D78094345

Pulled By: xuzhao9

fbshipit-source-id: 23c8a6fd311bfc654de9502c067dd3fdccac8152

manman-ren committed:
Add Gluon FA Blackwell for performance measurement (#268)

Summary:
Copy python/tutorials/gluon/01-attention-forward.py from OSS Version from PR7238

Pull Request resolved: https://github.com/pytorch-labs/tritonbench/pull/268

Reviewed By: xuzhao9

Differential Revision: D77901791

Pulled By: manman-ren

fbshipit-source-id: 4626816698fd55efc999f498660e02417b3b6b11

xuzhao9 committed:
Cleanup gbps metrics

Summary: gbps is a user-custom metric, not built-in metric, we are unifying its API here.

Reviewed By: FindHao, adamomainz

Differential Revision: D78163498

fbshipit-source-id: 3bdca98094b2ad1b2e1041cfb593c538562a8dc8

xuzhao9 committed:
Clean up ncu_trace and ncu metrics

Summary: Remove `ncu_trace` metric as we have decided to integrate NCU metric analysis within the framework. Now we always save the replay file, analyze the replay file, and aggregate the metrics.

Reviewed By: FindHao

Differential Revision: D78164136

fbshipit-source-id: 399106cb92fabd2e708e94168e1fde85ef17c627

manman-ren committed:
Add gluon persistent implementation (#278)

Summary:
Copied from OSS
Run with WITH_GLUON=1

Pull Request resolved: https://github.com/pytorch-labs/tritonbench/pull/278

Reviewed By: njriasan

Differential Revision: D78180963

Pulled By: manman-ren

fbshipit-source-id: 4cec69405047e0899c046ef58956fffcbb50b957

yf225 committed:
Fix accuracy check for flash_attention kernels (#280)

Summary:
Stacked PRs:
 * https://github.com/pytorch-labs/tritonbench/issues/276
 * __->__#280

 --- --- ---

### Fix accuracy check for flash_attention kernels

Pull Request resolved: https://github.com/pytorch-labs/tritonbench/pull/280

Reviewed By: oulgen

Differential Revision: D78222794

Pulled By: yf225

fbshipit-source-id: 4c104b1defc84a5ffecbe274ea00c590f9463362

yf225 committed:
Add accuracy check and fixes for fp8_attention Triton kernels (#276)

Summary:
Stacked PRs:
 * https://github.com/pytorch-labs/tritonbench/issues/281
 * __->__#276

 --- --- ---

### Add accuracy check and fixes for fp8_attention Triton kernels

Pull Request resolved: https://github.com/pytorch-labs/tritonbench/pull/276

Reviewed By: xuzhao9

Differential Revision: D78183079

Pulled By: yf225

fbshipit-source-id: 0a2a46a120163cc0bcfc03a6a879129c028b1f2e

yf225 committed:
Allow customizing inputs for cross_entropy benchmark (#281)

Summary:
Stacked PRs:
 * https://github.com/pytorch-labs/tritonbench/issues/285
 * https://github.com/pytorch-labs/tritonbench/issues/284
 * https://github.com/pytorch-labs/tritonbench/issues/283
 * https://github.com/pytorch-labs/tritonbench/issues/282
 * __->__#281

 --- --- ---

### Allow customizing inputs for cross_entropy benchmark

Pull Request resolved: https://github.com/pytorch-labs/tritonbench/pull/281

Reviewed By: oulgen

Differential Revision: D78230194

Pulled By: yf225

fbshipit-source-id: 4717a97a440e5f78245e2da1da8609583b137012

yf225 committed:
Fix bug in average row display

Differential Revision: D78253862

Pull Request resolved: https://github.com/pytorch-labs/tritonbench/pull/282

yf225 committed:
Pass in weight tensor via kernel function arg for fused_linear_cross_entropy (#283)

yf225 committed:
Set baseline for ragged_attention (#284)

yf225 committed:
welford: adjust tolorance to make accuracy check pass (#285)

xuzhao9 committed:
[backends] Add Quack (#289)

yf225 committed:
Use AsyncTaskContext to replace tl.async_task usage (#288)

xuzhao9 committed:
[be] Reorganize logging dir (#287)

xuzhao9 committed:
[install] Fix quack install (#293)

drisspg committed:
Add more options to triton bench (#292)

stack-info: PR: https://github.com/pytorch-labs/tritonbench/pull/292, branch: drisspg/stack/1

htyu committed:
Update HSTU pin. (#294)

jialiangqu committed:
Add --list-metrics command to display available metrics (#297)

jialiangqu committed:
implement --list-backends functionality and integrate with --list-metrics (#301)

* add a util file to get operator level details like backends and metrics

* feat: implement --list-backends functionality and integrate with --list-metrics

- Add new --list-backends flag to list operator backends with labels and status indicators
- Create unified list_operator_details.py module to handle both metrics and backends listing
- Support combined output when both --list-metrics and --list-backends are used
- Show backend status indicators: [baseline], [disabled], [fwd_only], [no_ci]
- Integrate per-operator display showing both metrics and backends together
- Replace old list_metrics.py with new unified approach
- Utilize helper functions from operator_utils.py for backend retrieval
- Maintain backward compatibility for existing --list-metrics functionality

Resolves: https://github.com/pytorch-labs/tritonbench/issues/115 (--list-backend requirement)

njriasan committed:
Fix Warpspec Matmul to be compatible with OmniFm Shapes

Differential Revision: D77950060

Pull Request resolved: https://github.com/pytorch-labs/tritonbench/pull/298

xuzhao9 committed:
WIP: TritonBench for FA4

Differential Revision: D78096306

Pull Request resolved: https://github.com/pytorch-labs/tritonbench/pull/296

njriasan committed:
Add small block shapes to warpspec matmul configs

Differential Revision: D78059342

Pull Request resolved: https://github.com/pytorch-labs/tritonbench/pull/299

jialiangqu committed:
nit: Replace spaces with tabs in output formatting

jialiangqu committed:
nit: Use indentation constants instead of tabs for better maintainability

jackiexu1992 committed:
Export GDPA kernel impl to OSS (#302)

Summary:
- Export GDPA kernel to OSS
- Clean up some functionality and make sure the kernel only includes GDPA

Differential Revision: D78926734

Co-authored-by: Han Xu <hanxu@meta.com>

FindHao committed:
Fix diff-train errors (#315)

* fix mis-sync caused by diff-train

* remove comments

* fix lint

njriasan committed:
Extend blackwell_matmul_descriptor_persistent to support transposed input

Differential Revision: D78756024

Pull Request resolved: https://github.com/pytorch-labs/tritonbench/pull/300

njriasan committed:
Add an inductor generated _scale_mm to TritonBench

Differential Revision: D79263834

Pull Request resolved: https://github.com/pytorch-labs/tritonbench/pull/316

yf225 committed:
Fix lint due to PR #302 (#313)

stack-info: PR: https://github.com/pytorch-labs/tritonbench/pull/313, branch: yf225/stack/12

yf225 committed:
Fix CSV output bug (#314)

stack-info: PR: https://github.com/pytorch-labs/tritonbench/pull/314, branch: yf225/stack/13

FindHao committed:
[TritonParse] Enable Launch Tracing, Default Parsing, and Enhanced Docs (#305)

* Enhance TritonParse logging initialization to enable launch tracing. This change updates the logging setup to include the `enable_launch_trace` parameter for improved traceability during execution.

* update arg

* enable for oss code

* Enhance TritonParse utility functions with detailed docstrings. Added module-level documentation and improved function comments for `tritonparse_init` and `tritonparse_parse` to clarify their purpose and usage.

* fix lint

NikhilAPatel committed:
Add Flag to Benchmark Rowwise Scaling

Differential Revision: D79266264

Pull Request resolved: https://github.com/pytorch-labs/tritonbench/pull/317

yf225 committed:
Add --only-match-mode=prefix-with-baseline to enable specific impl prefix together with baseline (#319)

njriasan committed:
Allow strides larger than the shape

Differential Revision: D78783068

Pull Request resolved: https://github.com/pytorch-labs/tritonbench/pull/303

yf225 committed:
Fix metric order non-determinism (#321)

manman-ren committed:
fix OSS tritonbench issues

Differential Revision: D79385285

Pull Request resolved: https://github.com/pytorch-labs/tritonbench/pull/322

njriasan committed:
Update Tensor Descriptor implementation to handle various layouts

Differential Revision: D78790372

Pull Request resolved: https://github.com/pytorch-labs/tritonbench/pull/304

jialiangqu committed:
Add A/B testing command line arguments

Core A/B Testing Features:
- Add --side-a/-a and --side-b/-b params
- Validation for single operator requirement  
- Error messages for wrong usage

Example usage:
python run.py --op vector_add --side-a "--precision fp16" --side-b "--precision fp32"

jialiangqu committed:
Implement A/B testing core functionality

Core A/B Testing Features:
- Add config parsing with proper argument handling
- Implement run_ab_test function for dual execution
- Support global and operator-specific parameters
- Basic A/B test results display

Example usage:
python run.py --op flex_attention --side-a='--precision fp16 --max-autotune' --side-b='--precision bf16 --dynamic'

jialiangqu committed:
A/B testing results collection and comparison output

- Create dedicated ab_test.py module for A/B testing functions
- Add detailed results comparison with performance calculations  
- Support tabular output across multiple backends and input shapes
- Clean up run.py by moving A/B logic to separate module

jialiangqu committed:
Fix warmup parameter type to int in parser

The warmup parameter was missing type=int declaration, n_warmup = max(1, int(warmup / estimate_ms)) in do_bench will throw type errors

jialiangqu committed:
Reformat A/B testing output

Restructure output into three sections:
- Configuration Analysis: shows parameter differences between A and B
- Performance Summary: displays average improvements per backend/metric
- Detailed Comparison: shows all metrics in tabular format per input size

jialiangqu committed:
Enhance A/B testing error handling

Add a bit more try catch blocks

jialiangqu committed:
Add A/B testing documentation

jialiangqu committed:
fix pylint errors

cbrownaz24 committed:
Adding Stream-K persistent matmul kernel (#309)

Differential Revision: D77983954

Pull Request resolved: https://github.com/meta-pytorch/tritonbench/pull/324

jialiangqu committed:
Remove unnecessary comment

FindHao committed:
fix argument type issue (#328) (#328)

Summary:
Pull Request resolved: https://github.com/meta-pytorch/tritonbench/pull/328

code change is self explanatory

Reviewed By: jackiexu1992, hx2224

Differential Revision: D79736514

Co-authored-by: Jacky Zhou <junqingz@meta.com>

jananisriram committed:
Add relative and absolute tolerance as command-line arguments

Differential Revision: D80137287

Pull Request resolved: https://github.com/meta-pytorch/tritonbench/pull/329

davidberard98 committed:
Update repo owner (#331)

tritonbench was moved from pytorch-labs/tritonbench to
meta-pytorch/tritonbench. This should allow CI to run.

xuzhao9 committed:
Fix cuda time (#337)

jananisriram committed:
Add input loader module for fp8_gemm

Differential Revision: D80565971

Pull Request resolved: https://github.com/meta-pytorch/tritonbench/pull/336

njriasan committed:
Add Support for TLX matmul (#340) (#340)

Summary:

Adds a TLX matmul to TritonBench and guards it behind TLX support.

Differential Revision: D79178034

xuzhao9 committed:
[infra] Fix docker workflow (#343)

xuzhao9 committed:
Always reset dynamo (#347)

xuzhao9 committed:
Default sm_scale to None (#344) (#344)

Summary:

To make the behavior consistent with FA4: https://www.internalfb.com/code/fbsource/fbcode/ai_codesign/gen_ai/flash_attention_v2/benchmarks/benchmark_attn.py


In FA4 benchmark, sm_scale defaults to None, which further translates to ` 1.0 / math.sqrt(head_dim)`: https://www.internalfb.com/code/fbsource/[f03d88961ee4ecd3f4ee76736d7de904351d295c]/fbcode/ai_codesign/gen_ai/flash_attention_v2/flash_attn/cute/interface.py?lines=112

whereas Tritonbench tries to pin the sm_scale to 1.3.

Change Tritonbench behavior to set sm_scale to None by default.

Reviewed By: njriasan, jduprat

Differential Revision: D80652124

ZainRizvi committed:
Replace 'pytorch-labs' with 'meta-pytorch' in 1 files (#354)

jananisriram committed:
Move scaling logic to input generation

Differential Revision: D80716975

Pull Request resolved: https://github.com/meta-pytorch/tritonbench/pull/352

jananisriram committed:
Add amax as default per-tensor scaling factor for fp8_gemm benchmark

Differential Revision: D80577628

Pull Request resolved: https://github.com/meta-pytorch/tritonbench/pull/339

xuzhao9 committed:
[ci] Add debugging information (#353)

htyu committed:
Use non-opt as baseline

Differential Revision: D81044493

Pull Request resolved: https://github.com/meta-pytorch/tritonbench/pull/358

xuzhao9 committed:
Make the result consistent with FAv4 benchmark

Differential Revision: D80984604

Pull Request resolved: https://github.com/meta-pytorch/tritonbench/pull/356

manman-ren committed:
[GDPA] add tlx fwd for blackwell

Differential Revision: D79301971

Pull Request resolved: https://github.com/meta-pytorch/tritonbench/pull/318

manman-ren committed:
subtiling activation to reduce register pressure

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

xuzhao9 committed:
[parser] Rename `--iters` to `--rep` (#359)

manman-ren committed:
use vector fma etc

Summary: use new way of calculating fast_gelu * scale

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

htyu commented on tritonbench/operators/gdpa/gdpa_blackwell_tlx.py:
```diff
@@ -161,10 +217,42 @@ def tanh_approx_fp32(x):
+    p0_view = tlx.local_reinterpret(qk_view_1st, dtype)
+    tlx.local_store(p0_view, p0)  # , tlx.storage_kind.tmem)
+
+    qk_view_2nd = tlx.subslice(qk_view, HEAD_DIM // 2, HEAD_DIM // 2)
```
nit: make this a loop so that it looks shorter?

htyu commented on tritonbench/operators/gdpa/gdpa_blackwell_tlx.py:
```diff
@@ -161,10 +217,42 @@ def tanh_approx_fp32(x):
+    inner = _mul_f32x2(inner, x)
+    out = _fma_f32x2(x, tanh_approx_fp32(inner), x)
+    return out  # x * tanh((c1 * x * x + c0)*x) + x
+    # return tanh_approx_fp32(x)
```
nit: remove the comment?

htyu reviewed (approved):
LGTM! Can you also paste perf numbers?

njriasan committed:
Enable sleeping between trials (#363) (#363)

Summary:
Pull Request resolved: https://github.com/meta-pytorch/tritonbench/pull/363

Adds command line support for a `--sleep` argument that can be used to inject a sleep in between trials. This is potentially useful on Blackwell where due to the floating clock measurements sometimes favor a earlier running kernel.

While most of this should be covered by just specifying running for at least 2 seconds, this can help ensure a more consistent run.

Reviewed By: xuzhao9, FindHao

Differential Revision: D81152631

manman-ren committed:
fix data type to not hardcode to float16

Differential Revision: D81138002

Pull Request resolved: https://github.com/meta-pytorch/tritonbench/pull/361
